core/api/src/main/scala/org/virtuslab/ideprobe/Config.scala

```diff
@@ -52,17 +52,17 @@ final case class Config(source: ConfigObjectSource, fallback: Option[Config] = N
 object Config {
   val Empty = new Config(ConfigSource.empty)
 
-  def fromString(str: String) = new Config(ConfigSource.string(str))
+  def fromString(str: String) = new Config(ConfigSource.default(ConfigSource.string(str)))
```

Comment:
Oh... so `default` reads application.conf/reference.conf, and then adds the supplied param on the top of that?

Reply:
```
  /** A config source for the default loading process in Typesafe Config with a custom application config source.
    * Typesafe Config stacks `reference.conf` resources provided by libraries, the given file and system property
    * overrides, resolves them and merges them into a single config.
    *
    * This method is provided here to support use cases that previously depended on `ConfigFactory.load(config)`.
    * Creating a custom source by merging the layers manually is usually recommended as it makes the config priorities
    * more transparent.
    *
    * @param appSource
    *   the source providing the application config
    * @return
    *   a `ConfigObjectSource` for the default loading process in Typesafe Config with a custom application config
    *   source.
    */
  def default(appSource: ConfigObjectSource): ConfigObjectSource =
    ConfigObjectSource(appSource.config().flatMap(ConfigFactoryWrapper.load))
```
TL;DR - it treats it's parameter (which is `appSource: ConfigObjectSource`) as the config with higher priority (as typesafe config normally does for the `application.conf`) and merges it with `reference.conf` (lower priority) if needed